### PR TITLE
Header matchers

### PR DESCRIPTION
**Pull request by alexeyshockov**

### CODE REVIEW

**Comment by lstrojny** [comment]
Couple of comments:
 - You need to extend the `ExtractorFactory` and include a `createHeaderValueExtractor` and a `createHeaderExistsExtractor()`. These return the values, nothing more.
 - If you want to use contains semantics then, you can pass e.g. a regex matcher (or we could implement a simple contains matcher as well)

File: src/InterNations/Component/HttpMock/Expectation.php [hunk 1]
```diff
@@ -55,6 +56,28 @@ public function methodIs($matcher)
...
+        );
+
+        return $this;
+    }
+
+    public function headerIs($name, $value)
+    {
+        $this->appendMatcher(
+            $this->matcherFactory->closure(function (Request $request) use ($name, $value) {
+                return $request->headers->contains($name, $value);
```
Comment (lstrojny) [inline src/InterNations/Component/HttpMock/Expectation.php]: Internally this does an in_array() check without the third parameter begin set to true, so it’s `==` not `===`. Can you rather do `in_array($value, $request->headers->get($name))`?

**Comment by franzliedke** [comment]
@lstrojny Thanks for this awesome library! It has helped me greatly in refactoring some third-party HTTP communication code to a better, object-oriented API.

In doing so, I've been using the `callback` matcher as well in order to ensure the existence of header values. I came here wanting to ask for a built-in matcher for this pretty common use-case, and found this PR. How can I help in getting this finished, merged and released? :)

**Comment by lstrojny** [comment]
@franzliedke glad to hear the library helped you. Any yes please, feel free to carry this PR forward, happy to get this merged.

**Comment by lstrojny** [comment]
Closing

**Pull request closed without merging**